### PR TITLE
feat: Bb 257 load and export nmp data to crops form module

### PR DESCRIPTION
**Pull request by GDamaso**
# Description

Crops are now loaded and exported in/from BetterBerries and NMP calc.

Some minor enhancements/simplifications were made to CropsInfo and CropsList so I could work with it and get the data I needed for building farmDetails and converting it to .nmp afterwards.

To test things out, I suggest creating a .nmp file, loading it to BB and seeing if it loads properly. Then create an .nmp in BB and try to load it to NMP calc.

[NMP Calculator](https://nmp.apps.nrs.gov.bc.ca/)

## Type of changes

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

### BB File in NMP - Fields
image

### BB File in NMP - Crops
image

### BB - Crops

ESLint would move back to a single line a function that was broken down from Prettier, throwing a max-line error.
ESlint config was updated to work with prettier, which resulted in a lot of formatting, which is why there seems
to be so many line changes in this PR. 

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-sustainment-capstone-2024-271-frontend.apps.silver.devops.gov.bc.ca)
- [Backend](https://nr-sustainment-capstone-2024-271-backend.apps.silver.devops.gov.bc.ca/api/health)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-sustainment-capstone-2024/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-sustainment-capstone-2024/actions/workflows/merge.yml)

### CODE REVIEW

**Review by GDamaso — COMMENTED** [review]
:ok_hand:

File: frontend/src/Commons/Forms/InputModules/Crops/CropsList.tsx [hunk 1]
```diff
@@ -29,7 +29,7 @@ const CropsList: FC<CropsListComponentProps> = ({ field }) => (
               <h2>Crop</h2>
               <h2>{crop.id + 1}</h2>
             </div>
-            <p key={crop.cropId}>{crop.cropId}</p>
+            <p key={crop.cropId}>{crop.cropId === '75' ? 'Blackberry' : 'Raspberry'}</p>
```
Comment (kcaparas1630) [inline frontend/src/Commons/Forms/InputModules/Crops/CropsList.tsx:32]: Do you mean Blueberry?

Comment (GDamaso) [reply frontend/src/Commons/Forms/InputModules/Crops/CropsList.tsx:32]: Yep

**Review by kcaparas1630 — APPROVED** [review]
Tested it out, and it works perfectly fine. I don't see anything wrong with the code too. I played with both nmp application and the better berry, and I must say I'm excited on where we're headed. Great work!

**Pull request merged**